lighthouse-core/test/computed/critical-request-chains-test.js

```diff
@@ -337,13 +360,15 @@ describe('CriticalRequestChain gatherer: extractChain function', () => {
     });
   });
 
-  it('handles non-existent nodes when building the tree', () => {
-    const networkRecords = mockTracingData([HIGH, HIGH], [[0, 1]]);
-    const mainResource = networkRecords[0];
-
-    // Reverse the records so we force nodes to be made early.
-    networkRecords.reverse();
-    const criticalChains = CriticalRequestChains.extractChain(networkRecords, mainResource);
+  it('handles non-existent nodes when building the tree', async () => {
+    const {networkRecords, criticalChains} = await createChainsFromMockRecords(
+      [HIGH, HIGH],
+      [[0, 1]],
+      networkRecords => {
+        // Reverse the records so we force nodes to be made early.
```

Comment:
I don't think this is testing the same thing anymore, maybe we can convert this to a test for the BFS-issue we're uncovering?